src/components/connection-method-selection-dialog.tsx

```diff
@@ -0,0 +1,132 @@
+    description: 'Using Keplr browser extension',
+  },
+  {
+    type: LikeCoinWalletConnectorMethodType.KeplrMobile,
+    name: 'Keplr Mobile',
+    tier: 2,
+    isMobileOk: true,
+    url: 'https://keplr.app/app',
+    description: 'Scan QR code with Keplr Mobile app',
+  },
+  {
+    type: LikeCoinWalletConnectorMethodType.LikerId,
+    name: 'Liker ID',
+    tier: 2,
+    isMobileOk: true,
+    url: 'https://liker.land/getapp',
+    description: 'Scan QR code with Liker Land app',
+  },
+  {
+    type: LikeCoinWalletConnectorMethodType.Cosmostation,
+    name: 'Cosmostation',
+    tier: 2,
+    isMobileOk: false,
+    url:
+      'https://chrome.google.com/webstore/detail/cosmostation/fpkhgmpbidmiogeglndfbkegfdlnajnf',
+    description: 'Using Cosmostation browser extension',
+  },
+].reduce(
+  (map, method) => {
+    map[method.type] = method;
+    return map;
+  },
+  {} as {
+    [type in LikeCoinWalletConnectorMethodType]: LikeCoinWalletConnectorMethod;
+  }
+);
+
+export interface ConnectionMethodSelectionDialogProps
+  extends HTMLAttributes<HTMLDivElement> {
+  methods: LikeCoinWalletConnectorMethodType[];
+  onClose?: () => void;
+  onConnect?: (method: LikeCoinWalletConnectorMethodType) => void;
+}
+
+/**
+ * Connection Method Selection Dialog
+ */
+export const ConnectionMethodSelectionDialog: FC<ConnectionMethodSelectionDialogProps> = ({
+  methods,
+  onClose,
+  onConnect,
+}) => {
+  const [isDialogOpen, setDialogOpen] = React.useState(true);
+
+  const isMobile = React.useMemo(isMobileDevice, []);
+
+  const tieredConnectionMethods = React.useMemo(() => {
+    const tieredMethods = methods
+      .filter(type => {
+        const method = connectionMethodMap[type];
+        return !!method && (!isMobile || (isMobile && method.isMobileOk));
+      })
+      .map(type => connectionMethodMap[type])
+      .reduce((tieredMethods, method) => {
+        if (!tieredMethods[method.tier]) {
+          tieredMethods[method.tier] = new Array<
+            LikeCoinWalletConnectorMethod
+          >();
+        }
+        tieredMethods[method.tier].push(method);
+        return tieredMethods;
+      }, {} as { [tier: string]: LikeCoinWalletConnectorMethod[] });
+    return Object.keys(tieredMethods)
+      .sort()
+      .map(key => tieredMethods[key]);
+  }, [methods, isMobile]);
+
+  function closeDialog() {
+    setDialogOpen(false);
+    if (onClose) onClose();
+  }
+
+  return (
+    <Dialog isOpen={isDialogOpen} onClose={closeDialog}>
+      <h1 className="lk-flex lk-items-center lk-gap-x-[12px] lk-text-like-green lk-font-bold">
+        <SignInIcon className="lk-w-[20px] lk-h-[20px] lk-shrink-0" />
+        <span>Connect a wallet</span>
+      </h1>
+      {isMobile && (
+        <Alert className="lk-mt-[24px]">
+          <p>
+            WalletConnect in mobile is an experimental feature, please visit
+            this site on desktop for a better experience.
+          </p>
+        </Alert>
```

Comment:
Should this warning be configurable off?

Reply:
Added config